[FLINK-26652][runtime] Makes Flink cluster not fail in case of the cleanup retries being exhausted (#19102)
apache/flink: docs/content/docs/deployment/overview.md

Once a job has reached a globally terminal state of either finished, failed or cancelled, the
external component resources associated with the job are then cleaned up. In the event of a
failure when cleaning up a resource, Flink will attempt to retry the cleanup. You can
[configure]({{< ref "docs/deployment/config#retryable-cleanup" >}}) the retry strategy used. 

> Reaching the maximum number of retries without succeeding will leave the job in a dirty state. 

Reviewer: Do we have any documentation that explains the consequences of being in a dirty state?

Author: I updated the documentation under `Deployment / High Availability / Overview / JobResultStore` to also cover dirty entries with references to the folders.

Its artifacts (and its [JobResultStore]({{< ref "docs/deployment/ha/overview#jobresultstore" >}}) 
entry like in Application Mode) would need to be cleaned up manually. Restarting the very same 

> job (i.e. using the same job ID) would result in the retryable cleanup being picked up again 

Reviewer: ```suggestion
job (i.e. using the same job ID) will result in the cleanup being restarted 
```
